### DISCUSSION TITLE
Google API error: request denied

### DISCUSSION THREAD

**Discussion by dp6ai** [chaskiq/chaskiq#426] [General]
I'm getting a `Google API error: request denied` error in the logs, I have set `DEFAULT_GEOCODER_SERVICE` and `GEOCODER_API_KEY`

EDIT: 
The full message is:
```(You must use an API key to authenticate each request to Google Maps Platform APIs. For additional information, please refer to http://g.co/dev/maps-no-account)```

But i have enabled both the geo location and maps api in GCP

I'm getting this is the Chaskiq logs even with the messenger disabled. 

Any pointers would be appreciated.

Thanks

**Comment by michelson** [comment]
what value are you using for DEFAULT_GEOCODER_SERVICE ?
it seems that the google maps key is invalid ?

**Reply by dp6ai** [reply]
`google`

**Reply by dp6ai** [reply]
do i need to extend `GEOCODER_SERVICES` to include google in the array with name, api_key etc If thats right, where should i do that? in a new Heroku envar called `GEOCODER_SERVICES`

**Reply by dp6ai** [reply]
As i've just seen google is not part of the defaults so the lookup in `DEFAULT_GEOCODER_SERVICE` will return nil

**Reply by michelson** [reply]
you are right, if you want to add it and I will be happy to merge that. behind the scenes the ip lookup uses geocoder ruby gem

**Reply by dp6ai** [reply]
Will do, I have another change that switches the `aws-ses` gem for `mail-ses`, the original gem no longer respects `region` and so always fails for non `US-EAST-1` regions. I'll send a PR for that also, but feel free to reject.

**Reply by michelson** [reply]
really ? I'm using aws-ses with eu regions too , it seems that aws-ses is more maintained than mail-ses (last release 3 years ago)

**Reply by dp6ai** [reply]
OK. It was a long time ago i was having these issues and it looks like the gem has been updated to handle regions now

[regions](https://github.com/drewblas/aws-ses#connecting-to-a-server-from-another-region-)

BUT i'm not sure Chaskiq will handle this out of the box with the current set up, although i may be missing something.

**Reply by michelson** [reply]
it think you are right , this parameters will be needed on the environments/production.rb :
```
  :server => 'email.#{region}.amazonaws.com',
  :message_id_domain => '#{region}.amazonses.com'
```

**Reply by dp6ai** [reply]
@michelson I've just sent a few PR's feel free to ignore if you dont think they are suitable.
